package.json

```diff
@@ -29,10 +29,11 @@
     "node": ">= 16.20.1"
   },
   "scripts": {
-    "prepare": "npm run compile",
     "compile": "node-gyp rebuild",
```

Comment:
Can we add run the `install-zstd` script as part of prepare to make the dev-experience here a little cleaner? Would be nice to just hit `npm install`, have npm download and install ztd and do the compilation of the bindings for me.

Reply:
> Would be nice to just hit npm install, have npm download and install ztd and do the compilation of the bindings for me.

Yeah, ideally for npm packages you can just do `git clone && npm install && npm test` (aka `git clone && npm it`). 

> No - if we leave the prepare script, then it will download zstd for users when they run npm install.

But not for `npm install @mongodb-js/zstd`, which is the common case, right? Like, this would only affect users who run `npm install` inside the source tree for this package, and those users _probably_ want that behavior, no?

Reply:
I think maybe you misread the NPM docs there - it says that `INIT_CWD` is always the root directory of our package when our scripts are run.

`prepare` doesn't work either way.  From the `npm install` docs:

> If there is a binding.gyp file in the root of your package and you haven't defined your own install or preinstall scripts, npm will default the install command to compile using node-gyp via node-gyp rebuild

preinstall runs before prepare (which runs after install).  I also don't think we want preinstall hooks, because those do run when the package is installed.

fwiw all our native packages have this issue.  Maybe we could consider adding a pretest script or something to streamline the clone + install + test workflow Anna mentioned.